### ISSUE TITLE
Pre-built package doesn't work with Alpine

### ISSUE THREAD

**Issue by hairyhum** [build-trust/ockam#2760]
When trying to build `ockam_services` using:
```
FROM elixir:1.13.4-alpine AS build
RUN apk add --no-cache \
  build-base=0.5-r2 \
  git \
  npm \
  python2 \
  libc6-compat \
  gcompat

RUN cd <path/to>/ockam_services && mix do deps.get, deps.compile
```

It fails to build `ockam_vault_software` with the error:
```
** (RuntimeError) Failed to load pre-built ockam vault NIF: Failed to load NIF library: 'Error relocating /app/_build/prod/lib/ockam_vault_software/priv/linux_x86_64/native/libockam_elixir_ffi.so: __register_atfork: symbol not found'
```

Alpine does not have `glibc` by default and even adding `gcompat` doesn't seem to work here.
Linux prebuilt library might need different lib files depending on the distro.

**Comment by thomcc** [comment]
Is there a reason we're using alpine? It's kind of a pain in the ass distro to build for (dynamically linked musl is an odd configuration).

It's doable, to be sure though.

**Comment by hairyhum** [comment]
I'll try to use a different distro. Just highlighting that alpine, which is used by many people is not supported by the pre-built libs at the moment.

**Comment by thomcc** [comment]
Yeah, we need to build the rust code for `x86_64-unknown-linux-musl` with `-Ctarget-feature=-crt-static` (by default it would have `+crt-static`).

I'm not sure how well supported this configuration is. I know it's on the plan to make the `-musl` rust targets not imply `+crt-static` anymore in the future, but until then I think that's the best option? I'm not sure though.

**Comment by thomcc** [comment]
Anyway, I called out that alpine might be a problem in https://github.com/build-trust/ockam/issues/2704. It's a problem for Rust in general at the moment. Possible to support, but tricky.

**Comment by mrinalwadhwa** [comment]
I don't think we want to support this. We should instead just pick a base image that has glibc.
